### ISSUE TITLE
[nginx] IPv6 support?

### ISSUE THREAD

**Issue by emilstahl** [certbot/certbot#3640]
Currently the nginx plugin adds:
`listen 443 ssl; # managed by Certbot`

It should add the following for IPv6 support:

```
listen 443 ssl; # managed by Certbot
listen [::]:443 ssl; # managed by Certbot
```

~An option for HTTP 2 would be nice too, either by specifying `--http2` flag or just check nginx for compatibility.;~

```
listen 443 ssl http2;
listen [::]:443 ssl http2;
```

**Comment by bmw** [comment]
@emilstahl, thanks for making this issue. I created #3646 so we can trick the issues of IPv6 and HTTP2 separately.

**Comment by kirelagin** [comment]
Actually, it is not entirely safe. In older versions of nginx the `ipv6only` was `off` by default, so having these two lines you propose at the same time would break the config.

I’d say that `ipv6only=on` should be added to the `listen [::]:443` line, but, AFAIU, this option is Linux-only and I’m not sure how nginx will respond to it on other OSes.

**Comment by jkldgoefgkljefogeg** [comment]
`ipv6only=on` is the default.   https://nginx.org/en/docs/http/ngx_http_core_module.html#listen
For a server config with `listen [::]:80;`, certbox does not generate `listen [::]:443;` At least user should be warned IPv6 SSL was not enabled.

**Comment by jplatte** [comment]
@bmw So what's the current state of this? You closed this issue in #5178, but then seem to have reverted your changes. Should this issue be re-opened?

**Comment by ohemorange** [comment]
@jplatte: We then unreverted the changes. The latest version of Certbot supports IPv6; see some caveats that we plan to fix in https://github.com/certbot/certbot/issues/5438.
